### ISSUE TITLE
document the best way to get started

### ISSUE THREAD

**Issue by kbilsted** [GustavEikaas/easy-dotnet.nvim#194]
at least on windows, there are a couple of approaches to installing neovim. Chocolatey, WSL 2, Virtualbox+linux distro

can we make an easy to follow guide for newcommers being interested in using neovim for C#.. for someone who only knows sat visual studio. What is the tested/recommended approach.

**Comment by GustavEikaas** [comment]
Yeah that wouldnt be a dumb idea. When it comes to installing neovim I guess there is a lot of material out there already. But for getting started with C# in neovim, definitely!

When it comes to installing I prefer chocolatey because its so simple to upgrade. WSL also works nice if you want to work on Linux. But all in all it doesnt matter how you install it

**Comment by kbilsted** [comment]
Ok let me elaborate. If i install using choco i also need a c compiler or zig, ripgrep and other packages. 

What about other dependencies. Im thinking mason etc or are all just dependencies loaded by your plugin automatically by a package manager?

Further, are you using neovim from within vscode or plain?

How to best associate files with neovim loading your plugin? I assume you dont want it loaded for non cs ?

Maybe we can do a video where we setup my machine from scratch? :)

**Comment by GustavEikaas** [comment]
Ah I see. Doing a video would be cool, and probably helpful for a bunch of other people getting started.

Just to help anyone stumbling across this issue I will answer your questions


- I use zig compiler because it was the easiest to install
- easy-dotnet.nvim doesnt rely on Mason
- I would not recommend associating files to the plugin, as most of its features are buffer independent
- Im using this plugin from neovim, I dont know of anyone using it through vscode and cant guarantee it even works

**Comment by jorgevds** [comment]
Maybe a good thread to mention the version of neovim you need in order to install this plugin. I tried on 0.9.5 and wasn't able, but upgrading to 0.10.2 worked. AFAIK this plugin uses a neovim builtin called vim.iter which is only available from version 0.10 onwards, yet the version information is not listed anywhere in the documentation of this plugin.

**Comment by GustavEikaas** [comment]
@jorgevds Ah good catch, I wasnt aware that iter is not available in 0.95, its a recent change I will revert it. I want to target 0.95 aswell

**Comment by GustavEikaas** [comment]
@jorgevds 
I tried to remove all new features that isnt available in 0.9.5 but its gonna take quit a bit of work. Being worked on in #205.
I updated the readme in #206 to reflect the requirement

**Comment by jorgevds** [comment]
@GustavEikaas looking great. Thanks for taking the time to do this!

**Comment by tobiasvandriessel** [comment]
@GustavEikaas , you mentioned you use Arch with this plugin as well and this makes me wonder: does this plugin work well on Linux? Reading through the ReadMe makes me think that Windows is required for the MSBuild and some Nuget stuff; is that correct? 

I'm looking into doing C# .NET 9+ development on my linux laptop and I agree with what was said before in this discussion, a getting-started would help greatly (if possible)! I'm currently hitting some issues when opening a directory with C# files, that I think are caused by prerequisites of easy-dotnet.nvim not being met (but I'm not sure because I'm using [a batteries-included setup of nvim](https://github.com/khaneliman/khanelivim), which includes a lot of other plugins as well):
- `Error: dotnet server exited with code 1` 
- `[server stderr] Could not execute because the specified command or file was not found.`

**Comment by tobiasvandriessel** [comment]
EDIT: I re-read your comment and notice that you actually mention dotnet _tools_, so I'll look into that! Thanks for the clue 🙏  

Thank you for your response!
Ah okay, then I'll need to put some more time into identifying the cause of my issue; I found these snippets in the ReadMe and thought this would be the issue.

I _should_ have dotnet in my path. However, I use a Nix Shell and am still quite new to it, so there's probably something I'm doing wrong there. Thanks again for your help, I'm looking forward to using your plugin!

**Comment by tobiasvandriessel** [comment]
Yes, thanks a bunch 🙏 It works really well!
